src/server/cluster/cluster_family_test.cc

```diff
@@ -589,6 +589,42 @@ TEST_F(ClusterFamilyTest, ClusterConfigDeleteSlots) {
                                                   IntArg(0), "total_writes", Not(IntArg(0)))))));
 }
 
+// Fix issue #1302
+TEST_F(ClusterFamilyTest, ClusterConfigDeleteSlotsNoCrushOnShutdown) {
+  string config_template = R"json(
+      [
+        {
+          "slot_ranges": [
+            {
+              "start": 0,
+              "end": 16383
+            }
+          ],
+          "master": {
+            "id": "$0",
+            "ip": "10.0.0.1",
+            "port": 7000
+          },
+          "replicas": []
+        }
+      ])json";
+  string config = absl::Substitute(config_template, RunAdmin({"dflycluster", "myid"}).GetString());
+
+  EXPECT_EQ(RunAdmin({"dflycluster", "config", config}), "OK");
+
+  Run({"debug", "populate", "100000"});
+
+  EXPECT_THAT(RunAdmin({"dflycluster", "getslotinfo", "slots", "1", "2"}),
+              RespArray(ElementsAre(
+                  RespArray(ElementsAre(IntArg(1), "key_count", Not(IntArg(0)), "total_reads",
+                                        IntArg(0), "total_writes", Not(IntArg(0)))),
+                  RespArray(ElementsAre(IntArg(2), "key_count", Not(IntArg(0)), "total_reads",
+                                        IntArg(0), "total_writes", Not(IntArg(0)))))));
+
+  config = absl::Substitute(config_template, "abc");
+  EXPECT_EQ(RunAdmin({"dflycluster", "config", config}), "OK");
+}
```

Comment:
I'd add a comment here, maybe along the lines of:
"We remove all slots from current instance and the immediately shut down to test that we do not crash"